internal/tableflow/command_topic_enable.go

```diff
@@ -125,6 +137,18 @@ func (c *command) enable(cmd *cobra.Command, args []string) error {
 				Kind: "Managed",
 			},
 		}
+	} else if strings.ToUpper(storageType) == "AZUREDATALAKESTORAGEGEN2" {
+		if !cmd.Flags().Changed("provider-integration") || !cmd.Flags().Changed("storage-account-name") || !cmd.Flags().Changed("container-name") {
+			return fmt.Errorf("provider-integration, storage-account-name and container-name flags are required when storage-type is AzureDataLakeStorageGen2.")
+		}
+		createTopic.Spec.Storage = &tableflowv1.TableflowV1TableflowTopicSpecStorageOneOf{
+			TableflowV1AzureAdlsSpec: &tableflowv1.TableflowV1AzureAdlsSpec{
+				Kind:                  "AzureDataLakeStorageGen2",
+				StorageAccountName:    *tableflowv1.PtrString(storageAccountName),
+				ContainerName:         *tableflowv1.PtrString(containerName),
+				ProviderIntegrationId: *tableflowv1.PtrString(providerIntegration),
```

Comment:
Are these fields strings or string pointers? If they're strings, we could pass the variables in directly instead of creating a string pointer and then immediately dereferencing it.

Reply:
Got it. Let's change both. The `*tableflowv1.PtrString(...)` pattern isn't necessary there either